planet/src/main/webapp/js/itinerary.js

```diff
@@ -21,37 +39,104 @@ function renderStartingLocation() {
     }
 }
 
-async function renderEvents() {
-    const eventsResponse = await fetch('/update-event');
-    const events = await eventsResponse.json();
+// Add an event to the firebase realtime database
+// Todo: input validation, success/failure callbacks
+async function addEvent() {
+    const eventName = document.getElementById('add-event-name').value;
+    const eventAddress = document.getElementById('add-event-address').value;
+    const eventDuration = document.getElementById('add-event-duration').value;
+    const userId = firebase.auth().currentUser.uid; 
+    
+    // Set every event's opening hours to 8am - 5pm for now
+    const openingTime = TimeRange.getTimeInMinutes(8,0);
+    const closingTime = TimeRange.getTimeInMinutes(17,0);
+
+    const eventListRef = database.ref('users/' + userId + '/currentList');
+    // Get the order number by counting existing events
+    const snap = await eventListRef.once('value');
+    const order = snap.numChildren() + 1;
+
+    //Create a new event
+    const newEventRef = eventListRef.push();
+    newEventRef.set({
+        name: eventName,
+        address: eventAddress,
+        duration: eventDuration,
+        openingTime: openingTime,
+        closingTime: closingTime,
+        order: order,
+    });
+    closeForm();
+}
+
+// Check authentication status and render the list of events if the user has
+// sign in.
+// Todo: discuss with teammates to decide what to show if user is not signed in.
+firebase.auth().onAuthStateChanged(function(user) {
+  if (user) {
+    renderEvents('currentList');
+  } else {
+     console.log('Please sign in');
```

Comment:
Are you using this for debugging? Users won't typically see this, right?
indentation also looks off.

Reply:
I added a ToDo above this because I think we should prompt users to sign in, either by showing an alert box or redirecting them to the sign in page directly. We also need to decide if users can see the itinerary page at all if they are not signed in. Since we haven't really discussed these ideas yet, I was planning to work on the else statement later.
Thanks for reminding me about the indentation! I just fixed it.